src/test_fixture/sharing.rs

```diff
@@ -251,23 +275,26 @@ where
     }
 }
 
+#[async_trait]
 impl<F: Field> ValidateMalicious<F> for [Vec<MaliciousReplicated<F>>; 3] {
-    fn validate(&self, r: F) {
+    async fn validate(&self, r: F) {
         assert_eq!(self[0].len(), self[1].len());
         assert_eq!(self[0].len(), self[2].len());
 
         for (m0, (m1, m2)) in zip(self[0].iter(), zip(self[1].iter(), self[2].iter())) {
-            [m0, m1, m2].validate(r);
+            [m0, m1, m2].validate(r).await;
         }
     }
 }
 
+#[async_trait]
+#[allow(unused_must_use)]
```

Comment:
See above; you should suppress the warning at the call site, not here.

Reply:
The compiler is confusing between MaliciousValidator::validate and ValidateMalicious::validate. Let me rename ValidateMalicious::validate and then it should be ok